### ISSUE TITLE
Order in output CSS file

### ISSUE THREAD

**Issue by DarKDinDoN** [vuejs/vue-loader#140]
Hello

Let's assume the following components :

```javascript
// Vue DevTools would show :
<Root>
  <SubRoot>
    <EndOfTree>
  
// Component Root
<template>
  <div class=".root-css">
    <sub-root></sub-root>
  </div>
</template>
<script>
  import SubRoot from './SubRoot.vue'
  export default {
    name: 'Root',
    components: { SubRoot }
  }
</script>
<style>.root-css { background-color: blue }</style> // <--- CSS Specific to this component


// Component SubRoot
<template>
  <div class=".sub-root-css">
    <end-of-tree-css></end-of-tree-css>
  </div>
</template>
<script>
  import EndOfTree from './EndOfTree.vue'
  export default {
    name: 'SubRoot',
    components: { EndOfTree }
  }
</script>
<style>.sub-root-css { background-color: red }</style>  // <--- CSS Specific to this component


// Component EndOfTree
<template>
  <div class=".end-of-tree-css">EndOfTree</div>
</template>
<script>
  export default {
    name: 'EndOfTree',
  }
</script>
<style>.end-of-tree-css { background-color: green }</style>  // <--- CSS Specific to this component
```
The issue : the production build contains the extracted css of all components in the wrong order.

Output CSS file is : 
- EndOfTree CSS
- SubRoot CSS
- Root CSS

Desired ouput CSS file : 
- Root CSS
- SubRoot CSS
- EndOfTree CSS

This is especially annoying if I have some reset CSS on the root component.

Note : I use vuejs-templates/webpack and vuejs/vue-router

**Comment by yyx990803** [comment]
Can't reproduce - it outputs the expected order for me. Please provide a reproduction repo.

**Comment by zheeeng** [comment]
@DarKDinDoN I met the same issue and I find it is because of I imported vue-router configurations which import route components before importing vue main component `<app>` which is the role as your `<root>` component.

**Comment by christophemarois** [comment]
For future reference, @zheeeng fixed my problem. In practice, I had to change:

```javascript
import router from './router'
import 'my-custom-css'
```

to:
```javascript
import 'my-custom-css'
import router from './router'
```

**Comment by greeven** [comment]
Thanks @christophemarois for giving your solution with a code example. Was really easy and fast to find a solution because of that.

**Comment by soarpatriot** [comment]
Thanks @christophemarois

**Comment by chimmelb** [comment]
I came across this issue while using bootstrap and vue. Bootstrap overriding our own styles (things looked find in dev).

We fixed by `import`ing bootstrap first, then all the Vue stuff. 

```
// Top of file
import 'bootstrap/dist/css/bootstrap.min.css'
import 'bootstrap/dist/js/bootstrap.min.js'
```

It worked prior to Bootstrap 4 where we were including styles **inside App.vue**

```
<style lang="sass">
  @import ./assets/sass/bootstrap/bootstrap.scss
</style>
<style lang="sass">
  @import ./assets/sass/custom.scss
</style>
```

That order was respected. When we removed SASS and imported directly in `main.js`, putting external CSS first in the file fixed the loading order we saw on Production.
